webui/react/src/utils/data.ts

```diff
@@ -149,3 +149,31 @@ export const isEqual = (a: unknown, b: unknown): boolean => {
   if (a === b) return true;
   return JSON.stringify(a) === JSON.stringify(b);
 };
+
+// export const getSubObjectCopy = (obj: RawJson, accessors: string[]): any => {
+//   return accessors
+//     .reduce((acc, cur) => acc === undefined ? acc : acc[cur], obj);
+// };
+
+// returns the sub object access by accessors (by reference).
+export const getSubObject = (obj: RawJson, accessors: string[]): any => {
+  let curIdx = 0;
+  let cur = obj;
+  while (isObject(cur) && curIdx < accessors.length) {
+    cur = cur[accessors[curIdx]];
+    curIdx++;
+  }
+  return cur;
+};
```

Comment:
**issue:** if I am reading this right, if `obj` is `{ x: { y: 2} }` and I call `getSubObject(obj, [ 'x', 'y', 'z' ])` I believe it will return a `2` instead of `undefined`.